libvips/create/sdf.c

```diff
@@ -0,0 +1,234 @@
+/* make various signed distance fields
+ *
+ * 16/6/24
+ * 	- from xyz.c
+ */
+
+/*
+
+	This file is part of VIPS.
+
+	VIPS is free software; you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+	02110-1301  USA
+
+ */
+
+/*
+
+	These files are distributed with VIPS - http://www.vips.ecs.soton.ac.uk
+
+ */
+
+/*
+#define VIPS_DEBUG
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif /*HAVE_CONFIG_H*/
+#include <glib/gi18n-lib.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <vips/vips.h>
+#include <vips/internal.h>
+#include <vips/debug.h>
+
+#include "pcreate.h"
+
+typedef struct _VipsSdf VipsSdf;
+
+typedef float (*PointFn)(VipsSdf *, int x, int y);
+
+struct _VipsSdf {
+	VipsCreate parent_instance;
+
+	int width;
+	int height;
+	char *name;
+
+	double cx;
+	double cy;
+	double r;
+
+	PointFn point;
+
+};
+
+typedef VipsCreateClass VipsSdfClass;
+
+G_DEFINE_TYPE(VipsSdf, vips_sdf, VIPS_TYPE_CREATE);
+
+static float
+vips_sdf_circle(VipsSdf *sdf, int x, int y)
+{
+	return sqrt(x * x + y * y) - sdf->r;
```

Comment:
```suggestion
	return hypot(x, y) - sdf->r;
```

Reply:
I always forget about `hypot()`. We're C99 now, so it's even easier.

Reply:
According to: https://github.com/libvips/libvips/blob/312ec1c2671b080a81cfca09b77afb7314553e69/libvips/create/worley.c#L197-L203 
`sqrt(x * x + y * y)` might be actually faster for int arguments.

Reply:
That's true, though `x * x` will overflow for ints over 32767, unfortunately :(